### DISCUSSION TITLE
What is the best way to stream data from parquet files? (JDBC)

### DISCUSSION THREAD

**Discussion by myselfmohdsameer** [duckdb/duckdb#9428] [Q&A]
Hey,

My use case is to process records stored in some parquet files and then save them back, using JDBC API.

- If I import data using read_parquet into a table, it loads everything in memory and does not spill to temp_directory, then I am processing the records of the table in batches using offset and limit.
- If I do a SELECT * on parquet and make a result set, if I iterate this result set, memory keeps on increasing indefinitely, does not spill.

I saw that documentation suggests not to use prepared statements to insert large amounts of data into DuckDB, so my first method is slashed out only second remains but yet I am unsure whether its an issue or not.

Thus, my question is what is the advised method for streaming data from parquet files?

**Comment by severo** [comment]
Do you remember which method you ended up using?

**Comment by staticlibs** [comment]
When reading large Parquet files from JDBC it may better to set `jdbc_stream_results=true` connection parameter, then result set read from Parquet won't be materialized by default. To insert the records from Parquet file into DuckDB though JDBC the most performant way would be to use the Appender interface that can be opened with `Connection#createAppender()`.
